src/UrlChecker/CakeRouterChecker.php

```diff
@@ -0,0 +1,106 @@
+<?php
+/**
+ * CakePHP(tm) : Rapid Development Framework (https://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @link          https://cakephp.org CakePHP(tm) Project
+ * @since         1.0.0
+ * @license       https://opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Authentication\UrlChecker;
+
+use Cake\Routing\Exception\MissingRouteException;
+use Cake\Routing\Router;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Message\UriInterface;
+
+/**
+ * Checks if a request object contains a valid URL
+ */
+class CakeRouterChecker implements UrlCheckerInterface
+{
+    /**
+     * Default Options
+     *
+     * - `checkFullUrl` Whether or not to check the full request URI.
+     *
+     * @var array
+     */
+    protected $_defaultOptions = [
+        'checkFullUrl' => false
+    ];
+
+    /**
+     * {@inheritdoc}
+     */
+    public function check(ServerRequestInterface $request, $loginUrls, array $options = [])
+    {
+        $options = $this->_mergeDefaultOptions($options);
+        $url = $this->_getUrlFromRequest($request->getUri(), $options['checkFullUrl']);
+
+        // Support string loginUrls
+        if (is_string($loginUrls)) {
```

Comment:
Why? I thought the goal of this checker was to use the router features?

Reply:
@markstory thats a question for @lilHermit

Reply:
I also don't think that is necessary. If someone needs to support both they can implement a checker collection.

Reply:
I just added it because thought if it was a string its easy to compare but happy for it to fail